### DISCUSSION TITLE
How to use TeslaMate?

### DISCUSSION THREAD

**Discussion by lenlo** [teslamate-org/teslamate#1558] [Getting started]
I'm sorry, this is newbie question and I apologize if it has been asked before. I tried looking through the instructions, issues, and previous discussions, but I couldn't see it anywhere. My question is this: How do you get TeslaMate to do anything useful? I've set it up in a docker container as per the instructions and when I go to http://myhost:4000, I can see my car's current location and some basic stats, but where is everything else? There are two choices on the menu bar saying Geo-Fences and Settings, but no Dashboards option that I have seen in the screenshots.

If I go to https://myhost:3000, I get a "Welcome to Grafana" page that talks about Grafana in general, but nothing about TeslaMate. Logging in as admin/admin makes no difference.

I've tried both driving the car around for a bit to collect data as well as import my existing TeslaFi data files, but seen no change.

This is very bewildering. Any idea of what might be wrong? I've tried this on two different hosts, both an M1 Mac running macOS 11.2.3 and a little arm64 Linux system running Debian 10, but neither show anything else, nor did I notice any errors during installation. Docker is freshly installed with the latest & greatest version.

Any suggestions?

(I'm intimately familiar with both Linux and macOS, but I've never used either docker or grafana before, so it's possible that I've missed something.)

**Comment by 65sc02** [comment]
> On 20. Apr 2021, at 08:10, Lennart Lövstrand ***@***.***> wrote:
> 
> 
> I'm sorry, this is newbie question and I apologize if it has been asked before. I tried looking through the instructions, issues, and previous discussions, but I couldn't see it anywhere. My question is this: How do you get TeslaMate to do anything useful? I've set it up in a docker container as per the instructions and when I go to http://myhost:4000 <http://myhost:4000/>, I can see my car's current location and some basic stats, but where is everything else? There are two choices on the menu bar saying Geo-Fences and Settings, but no Dashboards option that I have seen in the screenshots.
> 
> If I go to https://myhost:3000 <https://myhost:3000/>, I get a "Welcome to Grafana" page that talks about Grafana in general, but nothing about TeslaMate. Logging in as admin/admin makes no difference.
> 
> 

That would be the right page. Mine looks like this:


and in the left side (circled in red) you see the links to TeslaMate. Are those missing in your Grafana?
hh

**Reply by lenlo** [reply]
Sorry, I don't think screenshot made it though, but this is what it looks like for me (assuming that you can see my screenshots, that is):

![Screenshot 2021-04-19 at 23 22 40](https://user-images.githubusercontent.com/4592527/115347655-87157a80-a166-11eb-9c34-6cd0eee300a5.png)
![Screenshot 2021-04-19 at 23 23 09](https://user-images.githubusercontent.com/4592527/115347669-8bda2e80-a166-11eb-8934-cf70c051572f.png)

**Comment by 65sc02** [comment]
> On 20. Apr 2021, at 08:27, Lennart Lövstrand ***@***.***> wrote:
> 
> Sorry, I don't think screenshot made it though,
> 

Ah, too bad. Yours came through.

> but this is what it looks like for me (assuming that you can see my screenshots, that is):
> 
> 

I see. You are missing the link section on the left side. But I don’t know why, sorry!

hh

**Comment by eliluong** [comment]
What does your docker-compose.yml look like? Also what does `docker container ls` show? It should show teslamate, postgres, mosquitto, and grafana, at minimum. `docker-compose logs teslamate` after starting the docker container can help, too.

**Reply by lenlo** [reply]
I got `docker-compose.yml` from https://docs.teslamate.org/docs/installation/docker/. `docker container ls` shows all of the above running.

I think I may have found the problem, though: I noticed that the Dashboards URL field was empty when I went to the Settings page of the main TeslaMate app (http://myserver:4000). On a whim, I entered http://myserver:3000 and suddenly the Dashboards menu showed up on the menu bar with all the different statistics choices. Shouldn't it have been prefilled, though? At a very least, it would have been good to have it documented...

**Reply by eliluong** [reply]
Oh, yea, you're right, I think I ran into the same problem, too and had to fill that in. Been a while so I don't recall. Maybe it is because some people run their custom Grafana but I think it should be part of the install process, too.
